scripts/restore-sanitised-database.sh

```diff
@@ -96,17 +97,20 @@ aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- \
   --document-name AWS-StartPortForwardingSessionToRemoteHost \
   --parameters host="$DB_TO_RESTORE_HOST",portNumber="$REMOTE_SANITISE_DB_PORT",localPortNumber="$LOCAL_SANITISE_DB_PORT" &
 
+SSM_PID=$!
 
 # Step 2: Download the file from S3 to a temporary location
 echo "Downloading the backup file from S3..."
-aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE aws s3 cp s3://$S3_BUCKET/$S3_PATH $TEMP_FILE || { echo "Failed to download the backup file"; exit 1; }
+aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE aws s3 cp s3://$S3_BUCKET/$sanitised-db $sanitised-db || { echo "Failed to download the backup file"; exit 1; }
 
 # Step 3: Restore the database from the downloaded file
 echo "Restoring the database from the backup file..."
-pg_restore -d $DB_TO_RESTORE_URI -v --clean $TEMP_FILE || { echo "Failed to restore the database"; exit 1; }
+pg_restore -d $DB_TO_RESTORE_URI -v --clean $sanitised-db || { echo "Failed to restore the database"; exit 1; }
 
 # Step 4: Clean up temporary file
 echo "Cleaning up temporary files..."
-rm -rf $TEMP_FILE || { echo "Failed to remove temporary file"; exit 1; }
+rm -rf $sanitised-db || { echo "Failed to remove temporary file"; }
+
+trap clean EXIT INT TERM
```

Comment:
This trap should be set immediately after grabbing the PID, otherwise it won't fire if any of those commands in between fail.